libs/ui-components/src/hooks/useTimeZones.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+type TimeZoneMap = Record<string, string>;
+
+// Hook allows us to only load the timeZones when they are being used
+const useTimeZones = () => {
+  const [zones, setZones] = React.useState<TimeZoneMap>({});
+
+  React.useEffect(() => {
+    const loadZones = async () => {
+      const timeZones = (await import('./timeZones')).default;
```

Comment:
we should have a try/catch around this remote call
also maybe we want to have loading indication which could get handy on slower connections

Reply:
Could this actually be considered a remote call? It's lazy-loading a local file, and I would expect it to be very fast and unlikely to fail.

I tested it simulating a "Good 3G" connection and the slowest I saw was around 50ms. Not sure it's worth showing a loader or that. WDYT?

Reply:
It is a remote call. The browser is fetching the file from our webserver. 
I agree that the loading time will be very small and the possibility of it failing is low too.

We can also address it in a followup if needed.